src/lua/fileffi.lua

```diff
@@ -339,6 +345,40 @@ local function createFileWrapper(wrapper)
     return file
 end
 
+local function createUvListener(port, cb)
+    if type(port) ~= 'number' then error 'port argument must be an unsigned number' end
+    if (port) == nil then error 'must provide a port' end
+    if cb == nil then error 'must provide a callback function' end
+
+    _callback = function(fifo)  end
+
+    if cb ~= nil then
+        if type(cb) ~= 'function' then error 'callback must be a function' end
+        _callback = function(fifo)
+            file = createFileWrapper(fifo)
+            cb(file)
+        end
+    end
+
+    _callback = ffi.cast('void (*)(LuaFile* fifo)', _callback)
+
+    local wrapper = C.uvFifoListener()
+
+    local listener = {
+        _type = "LuaServer",
+        _wrapper = wrapper,
+        _proxy = newproxy(),
+        _cb = _callback,
+        _port = port,
+        startListener = function(listener) C.startListener(listener._wrapper, listener._port, listener._cb)  end,
+        stopListener = function(listener) C.stopListener(listener._wrapper) end,
+    }
+    -- Use a proxy instead of doing this on the wrapper directly using ffi.gc, because of a bug in LuaJIT,
+    -- where circular references on finalizers using ffi.gc won't actually collect anything.
+    debug.setmetatable(listener._proxy, { __gc = function()  C.stopListener(listener._wrapper) end })
```

Comment:
This should (1) call the deleter of the class instead, and (2) it should free the callback in Lua.